test/test-fs-event.c

```diff
@@ -389,8 +401,10 @@ TEST_IMPL(fs_event_watch_dir_recursive) {
 
   uv_run(loop, UV_RUN_DEFAULT);
 
-  ASSERT(fs_event_cb_called == fs_event_created + fs_event_removed);
-  ASSERT(close_cb_called == 2);
+  /* We do not need to check the number of fs_event_cb_called invocations
+   * because for the loop to terminate, we already know the number of create,
+   * remove and total fs_event_cb calls are correct. */
```

Comment:
I do mention this in the code that closes the handle: https://github.com/libuv/libuv/pull/589/files#diff-5737b5ae6c6acd2f41d43cd5a1ae5bbcR264  But I can add it here too as well.  I'll update the PR.